### PR TITLE
caret position tracking while editing input

### PR DESCRIPTION
**Pull request by viict**
Might have a better way to do this, but, I needed it in a hurry and it solves the problem for me as far as our QA have gonne so far.

### CODE REVIEW

**Comment by assisrafael** [comment]
Thank you for the pull request.
Track caret position is a very interesting feature! However you need to implement e2e tests for this pull request to be merged.
Are you familiar with [protractor](http://www.protractortest.org)?

**Comment by viict** [comment]
@assisrafael no, but, I will look into it and see if I can manage this test out

**Comment by assisrafael** [comment]
Thank you.

**Comment by juancarrey** [comment]
Hi, could this actually be used to provide a way of having default caret just before the decimal places ?
It is very annoying for users to always have to enter decimal places, and type 500 to put $5,00

**Comment by viict** [comment]
@juancarrey I'm currently trying to fix a minor problem and I will look into that right after I do the test case that I'm in dept with this project :snail:

**Comment by jayrmotta** [comment]
@viict I've tested your fork but for some reason I'm getting this exception:

```
TypeError: Cannot read property 'selectionStart' of undefined
    at Object.caretGet [as get] (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:93870:17)
    at Array.parser (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:93899:40)
    at NgModelController.$$parseAndValidate (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:38346:39)
    at NgModelController.$commitViewValue (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:38336:11)
    at http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:38479:15
    at Scope.$eval (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:28006:29)
    at Scope.$apply (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:28106:26)
    at NgModelController.$$debounceViewValueCommit (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:38478:15)
    at NgModelController.$setViewValue (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:38450:13)
    at HTMLInputElement.listener (http://10.0.1.236:3000/vendor.bundle.js?ec0f38d8a48f744511d3:34473:13) undefined
```

I've tracked the exception to [this line](https://github.com/viict/angular-input-masks/commit/aa6397085f5627c5e1c81bff0a8bc17f1efea1b9#diff-c9c4ac3ac992cf8890643441b6cd1db0R26) where you're trying to access `element[0]` but apparently it's not a collection.

**Comment by ianldgs** [comment]
One possible solution is this: https://github.com/bendrucker/angular-credit-cards/commit/b6c5644807ab01c7b765c7e9b0cd6ba5bbb5cee8

**Comment by jadir-junior** [comment]
for me that is not work

**Comment by vilasboas** [comment]
still not working for me

File: src/node_modules/mask-factory.js [hunk 1]
```diff
@@ -2,6 +2,35 @@
 
 module.exports = function maskFactory(maskDefinition) {
 	return function MaskDirective() {
+		var caret = {
+			set : function caretSet(element, pos) {
+				if (element[0].setSelectionRange) {
+					element[0].focus();
+					element[0].setSelectionRange(pos,pos);
```
Comment (wendellpalazzo) [inline src/node_modules/mask-factory.js]: Hi guys!

I had this problem too, but i modify a little this PR... placing the (setTimeout function) encapsulating ==> element[0].setSelectionRange(pos,pos);

like this:
setTimeout(function () {
	element[0].setSelectionRange(pos,pos);
}, 0);

this work for me... regards

**Comment by leuterio** [comment]
This worked for me:
<input autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"></input>

**Comment by assisrafael** [comment]
Closing due to inactivity.
Please reopen this PR if needed.

**Pull request closed without merging**